source/val/instruction.h

```diff
@@ -76,6 +76,15 @@ class Instruction {
     return inst_;
   }
 
+  // Casts the words belonging to the operand under |index| to |T| and returns.
+  template <typename T>
+  T GetOperandAs(size_t index) const {
+    const spv_parsed_operand_t& operand = operands_.at(index);
+    assert(operand.num_words * 4 >= sizeof(T));
+    assert(operand.offset + operand.num_words <= inst_.num_words);
+    return *reinterpret_cast<const T*>(&words_[operand.offset]);
```

Comment:
I could improve it to:
T x;
memcpy(&x, &words_[operand.offset], sizeof(T));
return x;